python/hail/table.py

```diff
@@ -1450,12 +1461,13 @@ def _process_joins(self, *exprs):
         broadcast_f = lambda left, data, jt: Table(left._jt.annotateGlobalJSON(data, jt))
         left, cleanup = process_joins(self, exprs, broadcast_f)
 
-        if left is not self:
+        if left is not self and (original_key is None and left.key is not None
+                                 or original_key is not None and (left.key is None or list(left.key) != original_key)):
```

Comment:
this all seems like a very complicated way of saying `original_key != left.key` but somehow `None` and `list` screw that up? Is there a way to make this less complicated? Or maybe if we had a `is_same_key` function? What exactly is a key? It's not `None or list(type)`, it seems like `None or ???` but I don't know what the ??? is.

Reply:
it's None or a StructExpression. I can simplify this though.